app/models/competition.rb

```diff
@@ -242,6 +246,30 @@ class Competition < ApplicationRecord
   validates :external_website, :external_registration_page, length: { maximum: MAX_URL_LENGTH }
   validates :contact, length: { maximum: MAX_MARKDOWN_LENGTH }
 
+  validate :validate_newcomer_reserved_spots
+  private def validate_newcomer_reserved_spots
+    return unless competitor_limit.present?
+
+    if newcomer_reserved_spots > 0 && !NEWCOMER_MONTH_ENABLED
+      errors.add(:newcomer_reserved_spots, 'newcomer competitions are not allowed at present')
+    end
+
+    max_newcomer_spots = (competitor_limit * MAX_NEWCOMER_SPOTS_RESERVED_FRACTION).to_i
+    if newcomer_reserved_spots > max_newcomer_spots
+      errors.add(:newcomer_reserved_spots, 'cant reserve more than 50% of spots for newcomers')
+    end
+  end
+
+  # TODO: Make sure this query is optimized
+  def newcomers_competing
+    registrations.competing_status_accepted.includes(:user).select { |registration| registration.user.newcomer? }
```

Comment:
Now that we're fully committed to V3, this can be `registrations.accepted...`

Reply:
Regarding optimization: That depends on what you're trying to achieve.
- If you really want to get the whole list of users, to access their properties (name, dob, email, whatever) or display them in a frontend _fully_, then this is as optimized as it can get
- If the only purpose of this method is to obtain a _count_ of newcomers, then you effectively discard all of the "hard work" of optimizing and preloading. A more effective way would need a `newcomer` **scope** on `User` to then do something along the lines of `registrations.accepted.joins(:user).merge(User.newcomer).count`

Reply:
Updated!